### DISCUSSION TITLE
How to view pre-release docs?

### DISCUSSION THREAD

**Discussion by jamesbraza** [astral-sh/ruff#4720] [Q&A] [answered]
The docs at https://beta.ruff.rs/docs/ don't support viewing pre-release docs (e.g. current `main` branch).

I am trying to review 9646bc7's `mkdocs` workflow [here](https://github.com/charliermarsh/ruff/actions/runs/5107950938/jobs/9181323858), and I don't see anywhere I can view the built docs.

In general, how do PR reviewers review docs PRs?  Besides looking at the PR diff, do they build the docs locally to view?

(I am wondering if there's a button I can hit, or a link I can click, and a built version of the docs be available)

**Comment by JonathanPlasse** [comment]
You can build and view the docs locally as described in `CONTRIBUTING.md`.
It should be possible to use dev container to do this online.

**Reply by jamesbraza** [reply]
Yeah I ended up just building locally.

Just wondering, what is dev container?  Is it something already added, or is it something that could be added?

**Reply by JonathanPlasse** [reply, marked as answer]
A devcontainer is a Visual Studio Code feature to set up the developing environment in a docker container.
You only need to have docker installed, all the setup will be done for you.
Github codespaces are devcontainer under the hood.

- This is available since #4678 has been merged.
